apps/web/src/components/Twitter/TwitterFollowingModal.tsx

```diff
@@ -0,0 +1,253 @@
+    graphql`
+      fragment TwitterFollowingModalFragment on Query
+      @refetchable(queryName: "TwitterFollowingModalQuery") {
+        socialConnections(
+          before: $twitterListBefore
+          last: $twitterListLast
+          socialAccountType: Twitter
+        ) @connection(key: "TwitterFollowingModal__socialConnections") {
+          edges {
+            node {
+              __typename
+              ... on SocialConnection {
+                __typename
+                galleryUser {
+                  ... on GalleryUser {
+                    id
+                    username
+                    bio
+                    ...FollowButtonUserFragment
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    `,
+    followingRef
+  );
+
+  const { hideModal } = useModalActions();
+
+  const handleClose = useCallback(() => {
+    hideModal();
+  }, [hideModal]);
+
+  const twitterFollowing = useMemo(() => {
+    const users = [];
+
+    for (const edge of followingPagination?.data?.socialConnections?.edges ?? []) {
+      if (edge?.node?.__typename === 'SocialConnection' && edge?.node?.galleryUser) {
+        users.push(edge?.node.galleryUser);
+      }
+    }
+
+    return users;
+  }, [followingPagination]);
+
+  const [followAll] = usePromisifiedMutation<TwitterFollowingModalMutation>(graphql`
+    mutation TwitterFollowingModalMutation {
+      followAllSocialConnections(accountType: Twitter) {
+        __typename
+        ... on FollowAllSocialConnectionsPayload {
+          __typename
+          viewer {
+            ... on Viewer {
+              __typename
+              user @required(action: THROW) {
+                __typename
+                id
+                dbid
+                following {
+                  id
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  `);
+  const reportError = useReportError();
+  const { pushToast } = useToastActions();
+
+  const handleFollowAll = useCallback(async () => {
+    try {
+      const updater: SelectorStoreUpdater<TwitterFollowingModalMutation['response']> = (
+        store,
+        response
+      ) => {
+        const followingUserIds = twitterFollowing.map((user) => user.id);
+
+        if (
+          response.followAllSocialConnections?.__typename === 'FollowAllSocialConnectionsPayload'
+        ) {
+          const currentId = query?.viewer?.user?.id;
+          if (!currentId) {
+            return;
+          }
+
+          const currentUserId = store.get(currentId);
+          if (!currentUserId) {
+            return;
+          }
+
+          for (const userId of followingUserIds) {
+            const user = store.get(userId);
+            const followers = user?.getLinkedRecords('followers');
+
+            if (followers) {
+              user?.setLinkedRecords([...followers, currentUserId], 'followers');
+            }
+          }
+        }
+      };
+
```

Comment:
@tbezman lmk if this is the right way to update all the follower store after the user clicked the `Follow All` button

https://user-images.githubusercontent.com/4480258/223043234-a427cf4e-ffd5-4ffe-979c-568145dd4eeb.mp4

Reply:
Yeah this is really nice